src/Dfe.PlanTech.Infrastructure.Data/CmsDbContext.cs

```diff
@@ -275,19 +278,42 @@ public async Task<List<T>> ToListAsync<T>(IQueryable<T> queryable, CancellationT
         var result = await _queryCacher.GetOrCreateAsyncWithCache(key, queryable,
             (q, ctoken) => q.FirstOrDefaultAsync(ctoken), cancellationToken);
 
-        //Attach the result to the DbContext change tracker.
-        //If the result was retrieved from the cache, or it is going to be used by another entity that _was_ retrieved from the cache,
-        //Then we need to ensure that EF Core is aware of it, otherwise it will not fix-up the navigations as it would normally
-        if (result != null)
-        {
-            AttachEntity(result);
-        }
+        AttachEntity(result);
 
         return result;
     }
 
-    public void AttachEntity<T>(T entity)
+    ///<summary>
+    ///Attachs the entity/marks it as "tracked" in the DbContext change tracker
+    ///</summary>
+    ///<remarks>
+    ///If the result was retrieved from the cache, or it is going to be used by another entity that _was_ retrieved from the cache,
+    ///Then we need to ensure that EF Core is aware of it, otherwise it will not fix-up the navigations as it would normally
+    ///</remarks>
+    private void AttachEntity<T>(T entity)
     {
+        //T can be any type, since FirstOrDefaultAsync or ToListAsync etc can return any type
+        //Therefore we need to check if there is actually a DbSet for the entity first, before attaching it
+        var isExistingDbSet = _dbSetTypes.Contains(typeof(T));
+
+        if(!isExistingDbSet){
+            //ADD LOGGING
+            return;
+        }
+
+        //Would ideally check if it's null, but it _has_ to be null to be a Db set entity
+        if(entity is default(T)){
```

Comment:
Yeah `default(T)` isn't a constant so it's failing but it's just so long winded.

I'm gonna see if it lets me get away with just the _one_ "issue" (null check not default check), if so I'm leaving it as null with a comment. Otherwise i'll change to this

Reply:
it seems to complain either way so I'd be tempted to do something like
```
if (!EqualityComparer<T>.Default.Equals(entity, default))
            base.Attach(entity!);
```